### PR TITLE
[MULTISIG] ImportTx unwraps MSig definition from UTXO allowing MSig x-chain transfer

### PR DESCRIPTION
**Pull request by pnowosie**
Follow up from https://github.com/chain4travel/caminogo/pull/217 allows Multisig owned UTXO to import into C-chain

### CODE REVIEW

File: plugin/evm/import_tx.go [hunk 1]
```diff
@@ -21,6 +19,9 @@ import (
 	"github.com/ava-labs/avalanchego/vms/components/avax"
 	"github.com/ava-labs/avalanchego/vms/components/verify"
 	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/coreth/core"
```
Comment (c4t-ag) [inline plugin/evm/import_tx.go]: They should go back into its own section on top as it was before

**Review by c4t-ag — CHANGES_REQUESTED** [review]

File: core/camino_alias_set.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+func (as *AliasSet) Add(aliasInfs ...verify.State) {
+	for _, ali := range aliasInfs {
+		alias, ok := ali.(*multisig.AliasWithNonce)
+		if !ok {
+			continue
+		}
+		old, has := (*as)[alias.ID]
+		if hasNewer := has && old.Nonce > alias.Nonce; hasNewer {
```
Comment (evlekht) [inline core/camino_alias_set.go]: omit hasNewer definition, just use expression

File: core/camino_alias_set.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	return alias, nil
+}
+
+func (as *AliasSet) Add(aliasInfs ...verify.State) {
+	for _, ali := range aliasInfs {
+		alias, ok := ali.(*multisig.AliasWithNonce)
+		if !ok {
+			continue
+		}
+		old, has := (*as)[alias.ID]
```
Comment (evlekht) [inline core/camino_alias_set.go]: inline this into if

File: core/camino_alias_set.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		return nil, database.ErrNotFound
+	}
+	return alias, nil
+}
+
+func (as *AliasSet) Add(aliasInfs ...verify.State) {
+	for _, ali := range aliasInfs {
+		alias, ok := ali.(*multisig.AliasWithNonce)
+		if !ok {
+			continue
```
Comment (evlekht) [inline core/camino_alias_set.go:27]: this should error, imo

File: core/camino_alias_set_test.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/components/multisig"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAliasSet(t *testing.T) {
```
Comment (evlekht) [inline core/camino_alias_set_test.go]:
```suggestion
func TestAliasSetGetMultisigAlias(t *testing.T) {
```

I'd suggest to split this into to unit tests - add and get

**Review by evlekht — CHANGES_REQUESTED** [review]

File: core/camino_alias_set.go (hunk 4 above)
Comment (pnowosie) [reply core/camino_alias_set.go:27]: I thought so, but then I should rollback the aliases that already went in(?)
...and then decided to ignore - WDYT?

File: core/camino_alias_set.go (hunk 3 above)
Comment (pnowosie) [reply core/camino_alias_set.go]: can't do - alias is used bellow

File: core/camino_alias_set.go (hunk 4 above)
Comment (evlekht) [reply core/camino_alias_set.go:27]: lets leave it as it is, k

File: plugin/evm/import_tx.go [hunk 6]
```diff
@@ -250,14 +259,30 @@ func (utx *UnsignedImportTx) SemanticVerify(
...
-		if err := vm.fx.VerifyTransfer(utx, in.In, cred, utxo.Out); err != nil {
+		if err := vm.fx.VerifyMultisigTransfer(utx, in.In, cred, utxo.Out, aliasSet); err != nil {
 			return fmt.Errorf("import tx transfer failed verification: %w", err)
 		}
 	}
 
 	return vm.conflicts(utx.InputUTXOs(), parent)
 }
 
+func unmarshallUtxo(codec codec.Manager, utxoBytes []byte) (*avax.UTXO, []verify.State, error) {
```
Comment (evlekht) [inline plugin/evm/import_tx.go]:
```suggestion
func unmarshalUTXO(codec codec.Manager, utxoBytes []byte) (*avax.UTXO, []verify.State, error) {
```

File: plugin/evm/import_tx.go [hunk 7]
```diff
@@ -250,14 +259,30 @@ func (utx *UnsignedImportTx) SemanticVerify(
...
+	var err error
+
+	wrappedUtxo := &avax.UTXOWithMSig{}
```
Comment (evlekht) [inline plugin/evm/import_tx.go]:
```suggestion
	wrappedUTXO := &avax.UTXOWithMSig{}
```

**Review by evlekht — CHANGES_REQUESTED** [review]

**Review by evlekht — APPROVED** [review]

**Review by c4t-ag — APPROVED** [review]

**Pull request merged**